src/main/java/com/github/maracas/roseau/diff/changes/JsonFormatter.java

```diff
@@ -0,0 +1,46 @@
+package com.github.maracas.roseau.diff.changes;
+
+import java.util.List;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.github.maracas.roseau.api.model.SourceLocation;
+
+public class JsonFormatter implements BreakinChangesFormatter {
+
+    /**
+     * Formats the list of breaking changes in JSON format
+     */
+    @Override
+    public String format(List<BreakingChange> breakingChanges) {
+        JSONArray jsonArray = new JSONArray();
+        
+        for (BreakingChange bc : breakingChanges) {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("element", bc.impactedSymbol().getQualifiedName());
+
+            jsonObject.put("oldPosition", createLocationJson(bc.impactedSymbol().getLocation()));
+
+            if (bc.newSymbol() != null) {
+                JSONObject newPosition = createLocationJson(bc.newSymbol().getLocation());
+                jsonObject.put("newPosition", newPosition);
+            } else {
+                jsonObject.put("newPosition", "");
```

Comment:
Do we keep a `newLocation` attribute in this case?

Reply:
Yes perhaps we could remove it. I'm not sure if having an empty string could be useful for the API or a report.

Reply:
An empty `newPosition` could mean that there's one but we couldn't infer it. I'd rather not include this attribute if there's none. Also, I'd stick with the `oldLocation`/`newLocation` terminology rather than `oldPosition`/`newPosition` :)